### ISSUE TITLE
wallet transfer and balance command appears to accept invalid wallet/key addresses.

### ISSUE THREAD

**Issue by dan-da** [maidsafe/safe-api#396]
I was wondering what would happen if I tried sending to an invalid recipient key.  

I start with 2 wallets, with balances.   Note that they end with 1y9e and cy9e, respectively.
```
$ safe wallet balance safe://hnyybypthxpghway6un8camdwikqpu3aftowi6kwtuhy9ueeabbeh1j551y9e
Wallet at "safe://hnyybypthxpghway6un8camdwikqpu3aftowi6kwtuhy9ueeabbeh1j551y9e" has a total balance of 21.143400000 safecoins

safe wallet balance safe://hnyybyxxehejryok44nx4quytbm36z59ptuq4btui98bpd8kn1mjd3hdocy9e
Wallet at "safe://hnyybyxxehejryok44nx4quytbm36z59ptuq4btui98bpd8kn1mjd3hdocy9e" has a total balance of 4.500000000 safecoins
```

Then transfer 1.5 Safecoin from 1y9e to cy9e, and succeeds as it should.

```
$ safe wallet transfer 1.5 --from safe://hnyybypthxpghway6un8camdwikqpu3aftowi6kwtuhy9ueeabbeh1j551y9e --to safe://hnyybyxxehejryok44nx4quytbm36z59ptuq4btui98bpd8kn1mjd3hdocy9e
Success. TX_ID: 17933436211355203536

danda@t61:~/dev/maidsafe/safe-api/safe-cli$ safe wallet balance safe://hnyybyxxehejryok44nx4quytbm36z59ptuq4btui98bpd8kn1mjd3hdocy9eWallet at "safe://hnyybyxxehejryok44nx4quytbm36z59ptuq4btui98bpd8kn1mjd3hdocy9e" has a total balance of 6.000000000 safecoins
```

Now transfer 1.5 Safecoin from 1y9e to dy9e.  dy9e does not (should not) exist.   This should fail, and it does.

```
$ safe wallet transfer 1.5 --from safe://hnyybypthxpghway6un8camdwikqpu3aftowi6kwtuhy9ueeabbeh1j551y9e --to safe://hnyybyxxehejryok44nx4quytbm36z59ptuq4btui98bpd8kn1mjd3hdody9e
[2020-01-21T23:04:25Z ERROR safe] safe-cli error: [Error] ContentError - No Wallet found at Wallet "safe://hnyybyxxehejryok44nx4quytbm36z59ptuq4btui98bpd8kn1mjd3hdody9e"
```

Now transfer 1.5 Safecoin from 1y9e to dyze.  cyze does not (should not) exist.   This should fail, but it succeeds, and the balance is the same as for cy9e, or even c111.

```
$ safe wallet transfer 1.5 --from safe://hnyybypthxpghway6un8camdwikqpu3aftowi6kwtuhy9ueeabbeh1j551y9e --to safe://hnyybyxxehejryok44nx4quytbm36z59ptuq4btui98bpd8kn1mjd3hdocyze
Success. TX_ID: 7763383359887524319

$ safe wallet balance safe://hnyybyxxehejryok44nx4quytbm36z59ptuq4btui98bpd8kn1mjd3hdocy9eWallet at "safe://hnyybyxxehejryok44nx4quytbm36z59ptuq4btui98bpd8kn1mjd3hdocy9e" has a total balance of 7.500000000 safecoins

$ safe wallet balance safe://hnyybyxxehejryok44nx4quytbm36z59ptuq4btui98bpd8kn1mjd3hdocyze
Wallet at "safe://hnyybyxxehejryok44nx4quytbm36z59ptuq4btui98bpd8kn1mjd3hdocyze" has a total balance of 7.500000000 safecoins

$ safe wallet balance safe://hnyybyxxehejryok44nx4quytbm36z59ptuq4btui98bpd8kn1mjd3hdoc111
Wallet at "safe://hnyybyxxehejryok44nx4quytbm36z59ptuq4btui98bpd8kn1mjd3hdoc111" has a total balance of 7.500000000 safecoins
```

It seems that the last 3 chars (y9e) are not part of the unique identifier.  It is curious though that it is the same suffix for both wallets, as it should change if it were a checksum.  And if it were a checksum, changing it to other values would result in validation error.  Regardless it is unexpected behavior that the last 3 hex values can be changed arbitrarily.

**Comment by bochaco** [comment]
@dan-da , I initially thought checksums were not really needed but as you point our here, I also started seeing this type of issues with urls so it seems we do need it, specially since wallets and safekeys have urls.

**Comment by S-Coyle** [comment]
Verified as resolved in PR #399 
To test I started off with a valid wallet to wallet transfer, then an invalid transfer where I changed the last 4 characters to an invalid wallet address. I then reduced this down to the last 3 characters, and finally to only the last character.
All of the tests that should have failed, did.
```
stephencoyle@Stephens-MacBook-Pro-2 ~/P/s/t/debug>
./safe wallet transfer 1.5 --from safe://hnyybyxmugjqakf73e3q38jp5q1imo4cmuifza1qckabgefgm4z9dt3pehy9e --to safe://hnyybyqr8xg994m4koh6ysa8pmeyg3utmywp7tpk56yuh7xkd8h4cy4wayy9e
Success. TX_ID: 13679741448462394851
[I] stephencoyle@Stephens-MacBook-Pro-2 ~/P/s/t/debug>
./safe wallet transfer 1.5 --from safe://hnyybyxmugjqakf73e3q38jp5q1imo4cmuifza1qckabgefgm4z9dt3pehy9e --to safe://hnyybyqr8xg994m4koh6ysa8pmeyg3utmywp7tpk56yuh7xkd8h4cy4waaaaa
[2020-01-27T17:30:35Z ERROR safe] safe-cli error: [Error] ContentError - No Wallet found at Wallet "safe://hnyybyqr8xg994m4koh6ysa8pmeyg3utmywp7tpk56yuh7xkd8h4cy4waaaaa"
[I] stephencoyle@Stephens-MacBook-Pro-2 ~/P/s/t/debug>
./safe wallet transfer 1.5 --from safe://hnyybyxmugjqakf73e3q38jp5q1imo4cmuifza1qckabgefgm4z9dt3pehy9e --to safe://hnyybyqr8xg994m4koh6ysa8pmeyg3utmywp7tpk56yuh7xkd8h4cy4wayaaa
[2020-01-27T17:31:41Z ERROR safe] safe-cli error: [Error] ContentError - No Wallet found at Wallet "safe://hnyybyqr8xg994m4koh6ysa8pmeyg3utmywp7tpk56yuh7xkd8h4cy4wayaaa"
[I] stephencoyle@Stephens-MacBook-Pro-2 ~/P/s/t/debug>
./safe wallet transfer 1.5 --from safe://hnyybyxmugjqakf73e3q38jp5q1imo4cmuifza1qckabgefgm4z9dt3pehy9e --to safe://hnyybyqr8xg994m4koh6ysa8pmeyg3utmywp7tpk56yuh7xkd8h4cy4wayy9a
[2020-01-27T17:31:47Z ERROR safe] safe-cli error: [Error] ContentError - No Wallet found at Wallet "safe://hnyybyqr8xg994m4koh6ysa8pmeyg3utmywp7tpk56yuh7xkd8h4cy4wayy9a"
```
